General/ProgrammaticCodeManipulation.md

```diff
@@ -0,0 +1,136 @@
+# Programmatic Code Manipulation (Pharo 11)
+
+In this article, we will present some code snippets that demonstrate how we can manipulate the source code (packages, classes, methods, variables, traits, etc.) programmatically, without using the SystemBrowser. Those techniques can be very useful for reflective operations, code generation, testing, etc.
+
+This article explains only those operations that **modify** the code. We will write a separate article on **querrying code** (looking for method senders, class references, pragma usages, etc.).
+
+## Packages
+
+### Creating a new package
+
+```st
+package := self packageOrganizer ensurePackage: 'PackageName'.
+```
+
+In Playground, you can use any object or class instead of `self`:
+
+```st
+package := Object packageOrganizer ensurePackage: 'PackageName'.
+```
+
+### Removing a package
+
+```st
+package removeFromSystem.
+```
+
+**Be careful!** In Pharo 11 and earlier, this will not properly remove classes from the system. So make sure to remove all classes before you remove the package.
+
+### Creating a package without installing it
+
+_(how to create an instance of a package without installing it into the image? - e.g., for testing)_
+
+## Classes
+
+### Creating a new class
+
+To create a new class, you can use the same code that appears in SystemBrowser. Here is an example with **Fluid** class definition.
+
+```st
+builder := Object << #MyClass
	slots: { };
	package: 'MyPackage'.
	
class := builder install.
+```
+
+The first statement will return an instance of `FluidClassBuilder`. The second statement will build the class, install it into the image, and assign it to the variable. If `MyPackage` does not exist, it will be created.
```

Comment:
This is false in P12. 

We had:
- FluidClassBuilder
- ShiftClassBuilder
- ShiftClassInstaller

FluidClassBuilder was created with the fluid syntax. When you wanted to install it, you had the create a ShiftClassInstaller that was delegating the building to the ShiftClassBuilder. 

But this had some flaws:
- The FluidClassBuilder could not manage to build a class with a name for the superclass instead of a class object. This was bad for code generation because it meant that the superclass had to exist when you wanted to define it (even if you did not build anything)
- The ShiftClassBuilder did not had a lot of API and some other API was complex because only used internally by the installer
- The API of both was kind of close but not exactly the same and this was confusing

In P12 I merged both of them inside the ShiftClassBuilder (There is no FluidClassBuilder anymore). And I tried to make sure both strength were kept.